javalib/src/main/scala/java/net/URI.scala

```diff
@@ -820,6 +820,14 @@ object URI {
     res
   }
 
+  /** Lower-cases host in `str` only if host exists. Used for hashing */
+  private def normalizeHost(str: String, host: js.UndefOr[String]): String = {
```

Comment:
I'm not sure this is correct. What if the authority part is `FoO.CoM@FoO.CoM`?

Reply:
That sounds right.
Addressed in https://github.com/scala-js/scala-js/pull/3690/commits/bba5ecc077c2298f087889cd701f6fe40339b537

Reply:
It feel this re-parses part of the URI again. Have you considered using the pre-parsed fields for this? (as in composing the hash code based on the present fields, rather than re-parsing and changing them)?

Reply:
Done in https://github.com/scala-js/scala-js/pull/3690/commits/cfdb791567791ced8e75ee70a407a7eb3d0d30ae
I think *before* is simpler than *after* though re-parsing.

Reply:
Personally I do prefer the new solution. @sjrd opinions?

Not sure we need the `List` with the `fold` rather than keeping mutable accumulation, but otherwise I like it.

Reply:
Yes, I'm not quite sure what the old version was, but the new one is good.

I would also revert to using the mutable accumulator. It was more readable *and* more efficient.